server/src/lib/authentication/whitelist/WhitelistHandler.ts

```diff
@@ -0,0 +1,50 @@
+import { AuthenticationSession } from "../../../../types/AuthenticationSession";
+import { AuthenticationSessionHandler } from "../../AuthenticationSessionHandler";
+import { IWhitelistHandler } from "./IWhitelistHandler";
+import { IUsersDatabase } from "../backends/IUsersDatabase";
+import { ServerVariables } from "../../ServerVariables";
+import Constants = require("../../../../../shared/constants");
+import Bluebird = require("bluebird");
+import express = require("express");
+import ipRangeCheck = require("ip-range-check");
+
+export const enum WhitelistValue {
+  NOT_WHITELISTED,
+  WHITELISTED,
+  WHITELISTED_AND_AUTHENTICATED_FIRSTFACTOR,
```

Comment:
I don't really understand why we have those two last values. Is there a way we can be asked to authenticate if we are already whitelisted? And if yes, what does it bring to the user?

Reply:
This ties back into the whitelist_policy conversation we have started above.

The last two values are used if you are attempting to hit non-whitelisted domain/resource. This is so it doesn't clash with your ability to view whitelisted and non-whitelisted domains/resources at the same time.